### PR TITLE
fix: bitstring for revocation registry

### PR DESCRIPTION
**Pull request by cristianIOHK**
### Description: 
This PR fixes the revocation registry verification using BitString and BitSet following the steps as the backend.

### Checklist: 
- [x] My PR follows the [contribution guidelines](https://github.com/input-output-hk/atala-prism-wallet-sdk-kmm/blob/main/CONTRIBUTING.md) of this project
- [x] My PR is free of third-party dependencies that don't comply with the [Allowlist](https://toc.hyperledger.org/governing-documents/allowed-third-party-license-policy.html#approved-licenses-for-allowlist)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked the PR title to follow the [conventional commit specification](https://www.conventionalcommits.org/en/v1.0.0/)

### CODE REVIEW

File: sampleapp/src/main/java/org/hyperledger/identus/walletsdk/sampleapp/Sdk.kt [hunk 1]
```diff
@@ -61,7 +61,9 @@ class Sdk {
 
         (pluto as PlutoImpl).start(context)
         agent.start()
-        agent.startFetchingMessages()
+        try {
+            agent.startFetchingMessages()
+        } catch (_: Exception) {}
```
Comment (curtis-h) [inline sampleapp/src/main/java/org/hyperledger/identus/walletsdk/sampleapp/Sdk.kt:66]: why are we swallowing exceptions here?

**Review by curtis-h — COMMENTED** [review]
nice, can we add some unit tests for the BitString class please?

File: edge-agent-sdk/src/commonTest/kotlin/org/hyperledger/identus/walletsdk/pollux/PolluxImplTest.kt [hunk 2]
```diff
@@ -1065,27 +1065,31 @@ class PolluxImplTest {
...
                 }
             }
         """
 
         pollux = PolluxImpl(apollo, castorMock, api)
-
         assertTrue(pollux.checkEncodedListRevoked(revocationRegistryJson, 1))
+        assertTrue(pollux.checkEncodedListRevoked(revocationRegistryJson, 2))
+        assertTrue(pollux.checkEncodedListRevoked(revocationRegistryJson, 3))
+        assertFalse(pollux.checkEncodedListRevoked(revocationRegistryJson, 4))
```
Comment (cristianIOHK) [inline edge-agent-sdk/src/commonTest/kotlin/org/hyperledger/identus/walletsdk/pollux/PolluxImplTest.kt:1092]: @curtis-h here I'm testing the pollux method that uses BitString to verify the credential revocation status.

File: sampleapp/src/main/java/org/hyperledger/identus/walletsdk/sampleapp/Sdk.kt (hunk 1 above)
Comment (cristianIOHK) [reply sampleapp/src/main/java/org/hyperledger/identus/walletsdk/sampleapp/Sdk.kt:66]: This is the sample app, the user of our SDK will decide how to handle exceptions here whether known or unknown.

File: edge-agent-sdk/src/commonTest/kotlin/org/hyperledger/identus/walletsdk/pollux/PolluxImplTest.kt (hunk 2 above)
Comment (curtis-h) [reply edge-agent-sdk/src/commonTest/kotlin/org/hyperledger/identus/walletsdk/pollux/PolluxImplTest.kt:1092]: yes, but this is test is more of an integration test, it lacks the direct assertion made possible with unit tests,  and there is no way of demonstrating the values given should yield the assertions being made

File: edge-agent-sdk/src/commonTest/kotlin/org/hyperledger/identus/walletsdk/pollux/utils/BitStringTest.kt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        }
+    }
+
+    @Test
+    fun `test BitString with some 1s`() {
+        val byteArray = byteArrayOf(128.toByte(), 1.toByte())
+        val bitSet = BitSet.valueOf(byteArray)
+        val bitString = BitString(bitSet, byteArray.size * 8)
+
+        assertTrue("Index 7 should be revoked", bitString.isRevoked(7))
```
Comment (curtis-h) [inline edge-agent-sdk/src/commonTest/kotlin/org/hyperledger/identus/walletsdk/pollux/utils/BitStringTest.kt:27]: how is this passing? isn't this test supposed to show position 0 and 15 as true?

**Review by elribonazo — APPROVED** [review]

**Pull request merged**